LibraBFT/Impl/Handle.agda

```diff
@@ -102,46 +102,33 @@ module LibraBFT.Impl.Handle where
- -- temporary scaffolding below.
-
- data Action (Msg : Set) : Set where
-   send : Msg → Action Msg
-
- action-send-injective : ∀ {Msg}{m m' : Msg} → send m ≡ send m' → m ≡ m'
- action-send-injective refl = refl
-
- msgToSend : {Msg : Set} → Action Msg → Msg
- msgToSend (send m) = m
-
- msgToSend≡ : ∀ {Msg x}{m : Msg} → m ≡ msgToSend x → send m ≡ x
- msgToSend≡ {_} {send m} {m} refl = refl
+ -- include things like logging, crashes, assertion failures, etc.
+ initWrapper : NodeId → GenesisInfo → RoundManager × List (LYT.Action NetworkMsg)
+ initWrapper nid g = ×-map₂ (List-map LYT.send) (initialRoundManagerAndMessages nid g)
 
  -- Note: the SystemModel allows anyone to receive any message sent, so intended recipient is ignored;
  -- it is included in the model only to facilitate future work on liveness properties, when we will need
  -- assumptions about message delivery between honest peers.
- outputToActions : RoundManager → Output → List (Action NetworkMsg)
- outputToActions rm (BroadcastProposal p) = List-map (const (Action.send (P p)))
+ outputToActions : RoundManager → Output → List (LYT.Action NetworkMsg)
+ outputToActions rm (BroadcastProposal p) = List-map (const (LYT.send (P p)))
                                                      (List-map proj₁
                                                                (kvm-toList (:vvAddressToValidatorInfo (₋esVerifier (₋rmEpochState (₋rmEC rm))))))
  outputToActions _  (LogErr x)            = []
- outputToActions _  (SendVote v toList)   = List-map (const (Action.send (V v))) toList
+ outputToActions _  (SendVote v toList)   = List-map (const (LYT.send (V v))) toList
 
- outputsToActions : ∀ {State} → List Output → List (Action NetworkMsg)
+ outputsToActions : ∀ {State} → List Output → List (LYT.Action NetworkMsg)
  outputsToActions {st} = concat ∘ List-map (outputToActions st)
 
- runHandler : RoundManager → LBFT Unit → RoundManager × List (Action NetworkMsg)
+ runHandler : RoundManager → LBFT Unit → RoundManager × List (LYT.Action NetworkMsg)
  runHandler st handler = ×-map₂ (outputsToActions {st}) (proj₂ (LBFT-run handler st))
 
  -- And ultimately, the all-knowing system layer only cares about the
  -- step function.
- peerStep : NodeId → NetworkMsg → Instant → RoundManager → RoundManager × List (Action NetworkMsg)
+ peerStep : NodeId → NetworkMsg → Instant → RoundManager → RoundManager × List (LYT.Action NetworkMsg)
  peerStep nid msg ts st = runHandler st (handle nid msg ts)
 
- -- This (temporary) wrapper bridges the gap between our (draft) concrete handler and
- -- the form required by the new system model, which does not (yet) support actions other
- -- than send.
- peerStepWrapper : NodeId → NetworkMsg → RoundManager → RoundManager × List NetworkMsg
- peerStepWrapper nid msg st = ×-map₂ (List-map msgToSend) (peerStep nid msg 0 st)
+ peerStepWrapper : NodeId → NetworkMsg → RoundManager → RoundManager × List (LYT.Action NetworkMsg)
```

Comment:
We don't do anything nontrivial with the timestamp at this stage: we just pass 0.  I think it would be worth commenting on that, and I don't really feel the need for both `peerStep` and `peerStepWrapper` (but keeping both is OK too).